libazureinit/src/imds.rs

```diff
@@ -32,64 +32,47 @@ pub async fn query_imds(
     if response.status().is_success() {
         let imds_body = response.text().await?;
 
-        return Ok(imds_body);
+        Ok(imds_body)
+    } else {
+        Err(Error::HttpStatus {
+            endpoint: url.to_owned(),
+            status: response.status(),
+        })
     }
-
-    println!(
-        "Get IMDS request failed with status code: {}",
-        response.status()
-    );
-    println!("{:?}", response.text().await);
-
-    Err(Box::from("Failed Get Call"))
 }
 
-pub fn get_ssh_keys(
-    imds_body: String,
-) -> Result<Vec<PublicKeys>, Box<dyn std::error::Error>> {
-    let data: Value = serde_json::from_str(&imds_body)
-        .expect("Failed to parse the IMDS JSON.");
+pub fn get_ssh_keys(imds_body: String) -> Result<Vec<PublicKeys>, Error> {
+    let data: Value = serde_json::from_str(&imds_body)?;
     let public_keys =
-        Vec::<PublicKeys>::deserialize(&data["compute"]["publicKeys"])
-            .expect("Failed to deserialize the public ssh keys.");
+        Vec::<PublicKeys>::deserialize(&data["compute"]["publicKeys"])?;
 
     Ok(public_keys)
 }
 
-pub fn get_username(
-    imds_body: String,
-) -> Result<String, Box<dyn std::error::Error>> {
-    let data: Value = serde_json::from_str(&imds_body)
-        .expect("Failed to parse the IMDS JSON.");
+pub fn get_username(imds_body: String) -> Result<String, Error> {
+    let data: Value = serde_json::from_str(&imds_body)?;
     let username =
-        String::deserialize(&data["compute"]["osProfile"]["adminUsername"])
-            .expect("Failed to deserialize the admin username.");
+        String::deserialize(&data["compute"]["osProfile"]["adminUsername"])?;
```

Comment:
Same. Should we then add it instead on the caller side, in `provision()`, when `get_username()` returned? Like:
```
let username = get_username(imds_body.clone()).with_context(|| "Failed to get the admin username.")?;
```